src/features/embalm/stepContent/hooks/useResurrection.ts

```diff
@@ -0,0 +1,83 @@
+import { useRadioGroup } from '@chakra-ui/react';
+import { minimumResurrection } from 'lib/constants';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import {
+  setResurrection,
+  setResurrectionRadioValue,
+  setCustomResurrectionDate,
+} from 'store/embalm/actions';
+import { useDispatch, useSelector } from 'store/index';
+
+/**
+ * Hook for managing resurrection selection
+ */
+export function useResurrection() {
+  const dispatch = useDispatch();
+  const {
+    resurrection,
+    resurrectionRadioValue: radioValue,
+    customResurrectionDate,
+  } = useSelector(x => x.embalmState);
+
+  // Although resurrection in ms is already stored in global state, we need another state to make it
+  // easier to display the value in the DatePicker. Resurrection and CustomResurrectionDate will often be
+  // managed in parallel.
+
+  const [error, setError] = useState<string | null>(null);
+
+  function handleRadioChange(nextValue: string) {
+    dispatch(setResurrectionRadioValue(nextValue));
+  }
+
+  function handleCustomDateChange(date: Date | null) {
+    dispatch(setCustomResurrectionDate(date));
+  }
+
+  // value and onChange are passed in to this hook instead of into a RadioGroup component.
+  // This is part of creating a custom radio component
+  const { getRadioProps } = useRadioGroup({
+    value: radioValue,
+    onChange: handleRadioChange,
+  });
+
+  // Update resurrection when the radio value changes
+  useEffect(() => {
+    if (radioValue !== 'Other') {
+      setCustomResurrectionDate(null);
+      if (radioValue === '') {
+        dispatch(setResurrection(0));
+      } else {
+        const [months] = radioValue === '' ? '0' : radioValue.split(' ');
+        const newResurrection = new Date();
+        newResurrection.setMonth(newResurrection.getMonth() + parseInt(months));
+        dispatch(setResurrection(newResurrection.getTime()));
```

Comment:
wondering if these resurrection setters should explicitly set the UTC timestamp, rather than parsing the resurrection time later, i.e.

```
const dateToUTC = (date: Date) => Math.floor(date.getTime() / 1000);
...
dispatch(setResurrection(dateToUTC(newResurrection.getTime())));
```

Reply:
It can go either way (i think), as `resurrection` is kept in sync with 'customResurrectionDate` which is a Date object so it needs the timestamp in ms. I'm pretty sure the timestamp is UTC time. But date/time can be confusing so maybe this needs revisited.